web/src/engine/websites/Dmzj.ts

```diff
@@ -0,0 +1,101 @@
+import { Tags } from '../Tags';
+import icon from './Dmzj.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchRequest } from '../FetchProvider';
+
+type APIResult<T> = {
+    errno: number,
+    errmsg: string,
+    data: T
+}
+
+type APIComicList = {
+    comicList: APIManga[]
+}
+
+type APIComicSingle = {
+    comicInfo: APIManga
+}
+
+type APIChapterInfo = {
+    chapterInfo: APIChapter
+}
+
+type APIManga = {
+    id: number,
+    name: string,
+    title: string
+    comicPy: string,
+    comic_py: string, //yes api is dumb like that
+    chapterList: APIChapter[]
+}
+
+type APIChapter = {
+    title: string,
+    data: {
+        chapter_id: number,
+        chapter_title: string
+    }[]
+    page_url_hd: string[],
+    page_url: string[]
+}
+
+type MangaID = {
+    comicPy: string,
+    comicId: number
+}
+
+type ChapterID = {
+    comicId: number,
+    chapterId: number
+}
+
+@Common.ImageAjax(true)
+export default class extends DecoratableMangaScraper {
+    public constructor() {
+        super('dmzj', `动漫之家(DMZJ)`, 'https://www.dmzj.com', Tags.Language.Chinese, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Source.Aggregator);
+    }
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string) {
+        return /^https?:\/\/www\.dmzj\.com\/info\/\S+\.html$/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const comicPy = url.match(/\/info\/(\S+)\.html$/)[1];
+        const request = new FetchRequest(new URL(`/api/v1/comic1/comic/detail?comic_py=${comicPy}`, this.URI).href);
+        const data = await FetchJSON<APIResult<APIComicSingle>>(request);
+        return new Manga(this, provider, JSON.stringify({ comicPy: data.data.comicInfo.comicPy, comicId: data.data.comicInfo.id }), data.data.comicInfo.title.trim());
```

Comment:
Is the `comicId` required as identifier?
It seems like `manga.comicId` is never used, only `comicPy` is used.